src/ApplicationLogs/LogReader.cs

```diff
@@ -37,30 +38,50 @@ public JObject GetApplicationLog(JArray _params)
             UInt256 hash = UInt256.Parse(_params[0].AsString());
             byte[] value = db.Get(ReadOptions.Default, hash.ToArray());
             if (value is null)
-                throw new RpcException(-100, "Unknown transaction");
-            return JObject.Parse(Encoding.UTF8.GetString(value));
+                throw new RpcException(-100, "Unknown transaction/blockhash");
+
+            //Additional optional "trigger" parameter to getapplicationlog for clients to be able to get just one execution result for a block.
+            if (_params.Count >= 2 && Enum.TryParse(_params[1].AsString(), true, out TriggerType trigger))
+            {
+                var raw = JObject.Parse(Encoding.UTF8.GetString(value));
+                var executions = raw["executions"] as JArray;
+                for (int i = 0; i < executions.Count;)
+                {
+                    if (!executions[i]["trigger"].AsString().Equals(trigger.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Does not support `All` ?
 https://github.com/neo-project/neo/blob/1d4db2cd4176390d18deeffc2aacc230a55950b0/src/neo/SmartContract/TriggerType.cs#L28

Reply:
I think it supports. If want to query `All`, just do not set the `trigger` parameter.

Reply:
I think the user may not know it, they're more likely to think that `TriggerType.All` is all 😅.

Reply:
I think that we should not allow `All`

Reply:
But why? It's the default anyway and I don't see any problem with specifying `All`.

Reply:
Because neo never execute the contract with `All`

Reply:
Actually, given the way triggers are defined I was expecting something more like
```
              if aer.Trigger&trig != 0 {
                      result = append(result, *aer)
              }
```
here (which is what we did in neo-go, for us specifying `All` is like not specifying anything). But I see now that ApplicationLogs plugin saves data using JSON in the DB and this doesn't allow for this trick (at least not that easy).

Reply:
Trigger it's a flag because it can be used for filters (Interops) but we always execute the contract as 'non flag'.